packages/graphback-core/src/db/buildModelTableMap.ts

```diff
@@ -0,0 +1,68 @@
+import { GraphQLField, GraphQLFieldMap, GraphQLObjectType } from "graphql";
+import { parseDbAnnotations } from '../annotations/parser';
+import { defaultColumnNameTransform, defaultTableNameTransform } from './defaultNameTransforms';
+import { getPrimaryKey } from './getPrimaryKey';
+
+/**
+ * Contains mapping information between GraphQL Model type and database table
+ * 
+ * - typeName: Original GraphQLObjectType name
+ * - tableName: Name of datase table
+ * - id: Indicates the primary key field
+ * - fieldMap: Object of key-value mapping between GraphQL fields and database columns.
+ */
+export interface ModelTableMap {
+  typeName: string
+  tableName: string
+  idField: string
+  fieldMap?: any
+}
+
+export const buildModelTableMap = (model: GraphQLObjectType): ModelTableMap => {
+  const primaryKeyField = getPrimaryKey(model);
+  const tableName = getTableName(model);
+  const fieldMap = mapFieldsToColumns(model.getFields());
+
+  return {
+    idField: primaryKeyField ? primaryKeyField.name : undefined,
+    typeName: model.name,
+    tableName,
+    fieldMap
+  }
+}
+
+function mapFieldsToColumns(fieldMap: GraphQLFieldMap<any, any>): any {
+  return Object.values(fieldMap).reduce((obj: any, field: GraphQLField<any, any>) => {
+    const columnName = getColumnName(field);
+
+    if (field.name !== columnName) {
+      obj[field.name] = columnName;
+    }
+
+    // TODO: Map relationship fields
+
+    return obj;
+  }, {});
+}
+
+export function getTableName(model: GraphQLObjectType): string {
+  let mappedTableName = defaultTableNameTransform(model.name, 'to-db');
+
+  const dbAnnotations = parseDbAnnotations(model);
+  if (dbAnnotations.name) {
+    mappedTableName = dbAnnotations.name;
+  }
+
+  return mappedTableName;
+}
+
+export function getColumnName(field: GraphQLField<any, any>): string {
+  let mappedColumnName = defaultColumnNameTransform(field.name, 'to-db');
```

Comment:
Why we doing this here. Can we skip transform by default when migration is happening. 
For example when we currently have only `model` annotation on plain type - What transforms this will produce? Do we need transforms at all. It seems to be overly complicated.

I kinda feel that we still need to control default transform but `  return lowerCaseFirstChar(name);` is not the best default value

Reply:
We are doing this here because our migration engine does a transformation of field names. IF for example you provide a model with `MyField: String` then `graphql-migrations` transforms this to column `myField`. In this way, we also need to do the same transformation here to find the default transformation value.

The migration package uses it from core:

https://github.com/aerogear/graphback/blob/0f1f48f238516eaa99f390721dd27c58ac0539a6/packages/graphql-migrations/src/migrate.ts#L1

So it is centrally controlled already.

Reply:
Lets keep it simple and remove transformation by default